File: backend/internal/domain/job.go
package domain

import "time"

// Status types and constants now defined in status.go (shared with all modules)

type PositionType string

const (
	PositionTypeEnterprise PositionType = "enterprise"
	PositionTypeTeaching   PositionType = "teaching"
)

type AbilityCategory string

const (
	AbilityCategoryKnowledge AbilityCategory = "knowledge"
	AbilityCategorySkill     AbilityCategory = "skill"
	AbilityCategoryQuality   AbilityCategory = "quality"
)

type AbilityPointSource string

const (
	AbilityPointSourcePublic AbilityPointSource = "public"
	AbilityPointSourceCustom AbilityPointSource = "custom"
)

// BatchStatus type and constants now defined in status.go (shared)

type CareerPosition struct {
	ID            string               `json:"id"`
	BatchID       *string              `json:"batchId,omitempty"`
	Name          string               `json:"name"`
	ShortName     *string              `json:"shortName,omitempty"`
	IndustryID    *string              `json:"industryId,omitempty"`
	MajorIDs      []string             `json:"majorIds"`
	MajorNames    []string             `json:"majorNames,omitempty"`
	PositionType  PositionType         `json:"positionType"`
	SalaryMin     *int                 `json:"salaryMin,omitempty"`
	SalaryMax     *int                 `json:"salaryMax,omitempty"`
	CoverImage    *string              `json:"coverImage,omitempty"`
	Description   *string              `json:"description,omitempty"`
	Requirements  []string             `json:"requirements"`
	CareerPath    *string              `json:"careerPath,omitempty"`
	Version       string               `json:"version"`
	Status        CareerPositionStatus `json:"status"`
	CreatedBy     string               `json:"createdBy"`
	Collaborators []string             `json:"collaborators"`
	CreatedAt     time.Time            `json:"createdAt"`
	UpdatedAt     time.Time            `json:"updatedAt"`
}

type PositionCertificate struct {
	ID               string  `json:"id"`
	CareerPositionID string  `json:"careerPositionId"`
	Name             string  `json:"name"`
	URL              *string `json:"url,omitempty"`
	Description      *string `json:"description,omitempty"`
	ImageURL         *string `json:"imageUrl,omitempty"`
}

type PositionResponsibility struct {
	ID               string  `json:"id"`
	CareerPositionID string  `json:"careerPositionId"`
	Name             string  `json:"name"`
	Description      *string `json:"description,omitempty"`
	SortOrder        int     `json:"sortOrder"`
}

type AbilityPoint struct {
	ID          string          `json:"id"`
	Name        string          `json:"name"`
	Description *string         `json:"description,omitempty"`
	Category    AbilityCategory `json:"category"`
	Code        *string         `json:"code,omitempty"`
	Domain      *string         `json:"domain,omitempty"`
	IsPublic    bool            `json:"isPublic"`
	CreatedAt   time.Time       `json:"createdAt"`
}

type PositionAbilityBinding struct {
	ID                string             `json:"id"`
	CareerPositionID  string             `json:"careerPositionId"`
	ResponsibilityID  string             `json:"responsibilityId"`
	AbilityPointID    string             `json:"abilityPointId"`
	Source            AbilityPointSource `json:"source"`
	Domain            *string            `json:"domain,omitempty"`
	RequiredLevel     string             `json:"requiredLevel"`
	RubricDescription *string            `json:"rubricDescription,omitempty"`
	Attributes        []string           `json:"attributes"`
	Weight            float64            `json:"weight"`
}

type AbilityDomain struct {
	ID               string   `json:"id"`
	CareerPositionID string   `json:"careerPositionId"`
	Name             string   `json:"name"`
	Description      *string  `json:"description,omitempty"`
	BindingIDs       []string `json:"bindingIds"`
	SortOrder        int      `json:"sortOrder"`
}

type JobBatch struct {
	ID             string      `json:"id"`
	Name           string      `json:"name"`
	Code           *string     `json:"code,omitempty"`
	OrgNodeID      *string     `json:"orgNodeId,omitempty"`
	MajorID        *string     `json:"majorId,omitempty"`
	MajorName      *string     `json:"majorName,omitempty"`
	WorkflowID     *string     `json:"workflowId,omitempty"`
	Status         BatchStatus `json:"status"`
	PositionCount  int         `json:"positionCount"`
	PublishedCount int         `json:"publishedCount"`
	PendingCount   int         `json:"pendingCount"`
	CreatedAt      time.Time   `json:"createdAt"`
	UpdatedAt      time.Time   `json:"updatedAt"`
}

type PositionRecommendation struct {
	ID               string    `json:"id"`
	MajorID          *string   `json:"majorId,omitempty"`
	MajorName        *string   `json:"majorName,omitempty"`
	CareerPositionID string    `json:"careerPositionId"`
	PositionType     string    `json:"positionType"`
	Reason           *string   `json:"reason,omitempty"`
	SortOrder        int       `json:"sortOrder"`
	IsEnabled        bool      `json:"isEnabled"`
	CreatedBy        string    `json:"createdBy"`
	CreatedAt        time.Time `json:"createdAt"`
	UpdatedAt        time.Time `json:"updatedAt"`
}

type LearnRoadStep struct {
	Name        string   `json:"name"`
	Description *string  `json:"description,omitempty"`
	ResourceIDs []string `json:"resourceIds,omitempty"`
}

type LearnRoad struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description *string   `json:"description,omitempty"`
	PositionIDs []string  `json:"positionIds"`
	Steps       JSONSlice `json:"steps,omitempty"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

File: packages/shared-types/src/job.ts
export interface CareerPosition {
  id: string
  batchId?: string
  name: string
  shortName?: string
  industryId?: string
  majorIds: string[]
  majorNames?: string[]
  positionType: "enterprise" | "teaching"
  salaryMin?: number
  salaryMax?: number
  coverImage?: string
  description?: string
  requirements: string[]
  careerPath?: string
  version: string
  status: "draft" | "pending" | "approved" | "rejected" | "published" | "archived"
  createdBy: string
  createdByName?: string
  collaborators: string[]
  collaboratorNames?: string[]
  favoriteCount?: number
  viewCount?: number
  createdAt: string
  updatedAt: string
}

export interface PositionCertificate {
  id: string
  careerPositionId: string
  certificateLibraryId: string
  name: string
  url?: string
  description?: string
  imageUrl?: string
}

export interface CertificateLibraryItem {
  id: string
  tenantId: string
  name: string
  url?: string
  description?: string
  imageUrl?: string
  createdAt: string
}

export interface PositionResponsibility {
  id: string
  careerPositionId: string
  name: string
  description?: string
  sortOrder: number
}

export interface AbilityPoint {
  id: string
  name: string
  code?: string
  description?: string
  category: "knowledge" | "skill" | "quality"
  attributes: string[]
  isPublic: boolean
  createdAt: string
}

export interface PositionAbilityBinding {
  id: string
  careerPositionId: string
  responsibilityId: string
  abilityPointId: string
  source: "public" | "custom"
  domain?: string
  requiredLevel: string
  rubricDescription?: string
  attributes: string[]
  weight: number
}

export interface AbilityDomain {
  id: string
  careerPositionId: string
  name: string
  description?: string
  bindingIds: string[]
  sortOrder: number
}

export interface JobBatch {
  id: string
  tenantId?: string
  name: string
  code?: string
  orgNodeId?: string
  majorId?: string
  majorName?: string // Deprecated: 仅兼容旧读取
  workflowId?: string
  status: "open" | "closed"
  positionCount: number
  publishedCount: number
  pendingCount: number
  createdAt: string
  updatedAt: string
}

export interface PositionRecommendation {
  id: string
  majorId?: string
  majorName?: string
  careerPositionId: string
  positionType: string
  reason?: string
  sortOrder: number
  isEnabled: boolean
  createdBy: string
  createdAt: string
  updatedAt: string
}

export interface BannerConfig {
  id: string
  title: string
  imageUrl: string
  linkUrl?: string
  sortOrder: number
  isEnabled: boolean
  createdAt: string
  updatedAt: string
}

export interface LearnRoadStep {
  name: string
  description?: string
  resourceIds?: string[]
  tasks?: { id: string; name: string }[]
}

export interface LearnRoad {
  id: string
  name: string
  description?: string
  positionIds: string[]
  steps?: LearnRoadStep[]
  createdAt: string
  updatedAt: string
}
